.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -732,6 +732,16 @@ private DateTimeResolutionResult ParseDuration(string text, DateObject reference
                         endTime = beginTime.AddSeconds(swiftSeconds);
                     }
 
+                    // Handle the "within xx seconds/minutes/hours" case
+                    // Should also handle the multiple duration case like P1DT8H
+                    // Set the beginTime equal to reference time for now
+                    prefixMatch = Config.WithinConnectorRegex.Match(beforeStr);
+                    if (prefixMatch.Success && prefixMatch.Length == beforeStr.Length)
+                    {
+                        mod = TimeTypeConstants.AFTER_MOD;
```

Comment:
Why this MOD?

Reply:
Datetime after now, should be AFTER_MOD?

Reply:
This is a range, so it needs to be inside the range. No need for any MOD.

Reply:
Fixed